proof/courier.go

```diff
@@ -91,12 +92,80 @@ type CourierCfg struct {
 	LocalArchive Archiver
 }
 
+// CourierConnStatus is an enum that represents the different states a courier
+// connection can be in.
+type CourierConnStatus int
+
+const (
+	// CourierConnStatusUnknown indicates that the connection status is
+	// unknown.
+	CourierConnStatusUnknown CourierConnStatus = iota
```

Comment:
non-blocking nit: may be good to have `unknown` after `disconnected`, to follow the enum pattern we have elsewhere / make comparison a bit safer.

Though with exhaustive switches I suppose the order doesn't matter.

Reply:
I think it's a good idea to have `unknown` as the first enum value, ensuring that the default for a `CourierConnStatus` variable is `unknown`. Which means that if we don't set the field, it won't default to something opinionated.